WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/WooShipping Package and Rate Selection/WooShippingAddPackageView.swift

```diff
@@ -0,0 +1,254 @@
+                    Spacer()
+                    Image(systemName: "chevron.up.chevron.down")
+                }
+                .padding()
+            }
+            .roundedBorder(cornerRadius: 8, lineColor: Color(.separator), lineWidth: 1)
+            HStack(spacing: 8) {
+                ForEach(WooShippingAddPackageDimensionView.DimensionType.allCases, id: \.self) { dimensionType in
+                    WooShippingAddPackageDimensionView(dimensionType: dimensionType, fieldValue: Binding(get: {
+                        return self.fieldValues[dimensionType] ?? ""
+                    }, set: { value in
+                        self.fieldValues[dimensionType] = value
+                    }))
+                }
+            }
+            Toggle(isOn: $showSaveTemplate) {
+                Text(Localization.saveNewPackageTemplate)
+                    .font(.subheadline)
+            }
+            if showSaveTemplate {
+                Button(Localization.savePackageTemplate) {
+                    // save template
+                }
+                .buttonStyle(SecondaryButtonStyle())
+            }
+        }
+        .padding()
+    }
+
+    private var carrierPackageView: some View {
+        // TODO: just a placeholder
+        Text("carrier")
+    }
+
+    private var savedPackageView: some View {
+        // TODO: just a placeholder
+        Text("saved")
+    }
+}
+
+struct WooShippingAddPackageDimensionView: View {
+    enum DimensionType: CaseIterable {
+        case length, width, height
+        var name: String {
+            switch self {
+            case .length:
+                return Localization.length
+            case .width:
+                return Localization.width
+            case .height:
+                return Localization.height
+            }
+        }
+    }
+
+    let dimensionType: DimensionType
+    @Binding var fieldValue: String
+    @FocusState var fieldFocused: Bool
+
+    var body: some View {
+        VStack {
+            HStack {
+                Text(dimensionType.name)
+                    .font(.subheadline)
+                Spacer()
+            }
+            HStack {
+                TextField("", text: $fieldValue)
+                    .keyboardType(.decimalPad)
+                    .font(.body)
+                    .focused($fieldFocused)
+                Text("cm")
+                    .font(.subheadline)
+                    .foregroundStyle(.secondary)
+            }
+            .padding()
+            .roundedBorder(cornerRadius: 8,
+                           lineColor: fieldFocused ? Color(UIColor.wooCommercePurple(.shade60)) : Color(.separator),
+                           lineWidth: fieldFocused ? 2 : 1)
+        }
+        .frame(minHeight: 48)
+    }
+}
+
+#Preview {
+    WooShippingAddPackageView()
+}
+
+extension WooShippingAddPackageView {
+    enum Localization {
+        static let addPackage = NSLocalizedString("wooShipping.createLabel.addPackage.title",
```

Comment:
@rachelmcr I have seen that we use "wooShipping.createLabel." prefix for creating shipping label, is it ok to continue using it?

Reply:
Yeah, that seems logical to me here!